packages/remark-grid-tables/src/index.js

```diff
@@ -448,6 +448,228 @@ function transformer (tree) {
+        grid[i][j + X].value = this.all(cell).join('\n\n').split('\n')
+
+        setHeigth(grid, i, j + X, grid[i][j + X].value)
+        setWeight(grid, i, j + X, cell.data.hProperties.colspan)
+
+        // If we have an empty 1x1 grid, we fill it up with a useless space
+        // Otherwise, it will not be parsed.
+        if (nbCols === nbRows && nbCols === 1 && !grid[0][0].value.join('\n')) {
+          grid[0][0].value = ' '
+          grid[0][0].w = 3
+        }
+
+        // Define the border of each cell
+        for (let x = 0; x < cell.data.hProperties.rowspan; x++) {
+          for (let y = 0; y < cell.data.hProperties.colspan; y++) {
+            // b attribute is for bottom
+            grid[i + x][j + X + y].b = x + 1 === cell.data.hProperties.rowspan
+            // r attribute is for right
+            grid[i + x][j + X + y].r = y + 1 === cell.data.hProperties.colspan
+
+            // set v if a cell has ever been define
+            grid[i + x][j + X + y].v = true
+          }
+        }
+      })
+      i++
+    })
+  })
+
+  // If they is 2 differents tableHeader, so the first one is a header and
+  // should be underlined
+  if (gridNode.children.length > 1) {
+    grid[gridNode.children[0].children.length - 1][0].H = true
+  }
+}
+
+function setSize (grid) {
+  // The idea is the max win
+
+  // Set the height of each column
+  grid.forEach(row => {
+    // Find the max
+    const mH = Math.max(...row.map(cell => cell.h))
+
+    // Set it to each cell
+    row.forEach(cell => { cell.h = mH })
+  })
+
+  // Set the weight of each row
+  grid[0].forEach((_, j) => {
+    // Find the max
+    const mW = Math.max(...grid.map(row => row[j].w))
+
+    // Set it to each cell
+    grid.forEach(row => { row[j].w = mW })
+  })
+}
+function generateBorders (grid, nbRows, nbCols, gridString) {
+  /** **** Create the borders *******/
+
+  // Create the first line
+  /*
+   * We have to create the first line manually because
+   * we process the borders from the attributes bottom
+   * and right of each cell. For the first line, their
+   * is no bottom nor right cell.
+   *
+   * We only need the right attribute of the first row's
+   * cells
+   */
+  let first = '+'
+  grid[0].forEach((cell, i) => {
+    first += '-'.repeat(cell.w)
+    first += cell.r || i === nbCols - 1 ? '+' : '-'
+  })
+
+  gridString.push(first)
+
+  grid.forEach((row, i) => {
+    let line = ''
+
+    // Cells lines
+    // The inner of the cell
+    line = '|'
+    row.forEach(cell => {
+      cell.y = gridString.length
+      cell.x = line.length + 1
+      line += ' '.repeat(cell.w)
+      line += cell.r ? '|' : ' '
+    })
+
+    // Add it until the text can fit
+    for (let t = 0; t < row[0].h; t++) {
+      gridString.push(line)
+    }
+
+    // "End" line
+    // It's the last line of the cell. Actually the border.
+    line = row[0].b ? '+' : '|'
+
+    row.forEach((cell, j) => {
+      let c = ' '
+
+      if (cell.b) {
+        if (row[0].H) {
```

Comment:
can you use more "obvious" name such as `isHeader`? idem for cell.b and cell.r etc.

Reply:
Rename it to `isHeader`, `hasBottom` and `hasRigth` ? 

I'm out of coke/chocolate now, so yes but i'm out for an hour. How ? As a new commit or an amend one ?

Thanks for the review. It's really appreciated.

Reply:
I think you should fixup the new commit to the old one.